ledger-core/v2/virtual/request/vstateunavailable.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package request
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/v2/log"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/injector"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/object"
+)
+
+type SMVStateUnavailable struct {
+	// input arguments
+	Meta    *payload.Meta
+	Payload *payload.VStateUnavailable
+}
+
+var dSMVStateUnavailableInstance smachine.StateMachineDeclaration = &dSMVStateUnavailable{}
+
+type dSMVStateUnavailable struct {
+	smachine.StateMachineDeclTemplate
+}
+
+func (*dSMVStateUnavailable) InjectDependencies(_ smachine.StateMachine, _ smachine.SlotLink, _ *injector.DependencyInjector) {
+}
+
+func (*dSMVStateUnavailable) GetInitStateFor(sm smachine.StateMachine) smachine.InitFunc {
+	s := sm.(*SMVStateUnavailable)
+	return s.Init
+}
+
+/* -------- Instance ------------- */
+
+func (s *SMVStateUnavailable) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return dSMVStateUnavailableInstance
+}
+
+func (s *SMVStateUnavailable) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	return ctx.Jump(s.stepProcess)
+}
+
+type noObjectErrorMsg struct {
+	*log.Msg  `txt:"There is no such object"`
+	Reference string
+}
+
+func (s *SMVStateUnavailable) stepProcess(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	catalog := object.Catalog{}
+	objectRef := s.Payload.Reference
+	sharedObjectState, ok := catalog.TryGet(ctx, objectRef)
+	if !ok {
+		ctx.Log().Error(noObjectErrorMsg{Reference: objectRef.String()}, nil)
+		return ctx.Stop()
+	}
+
+	setStateFunc := func(state *object.SharedState) {
+		if state.IsReady() {
+			panic(throw.IllegalState())
```

Comment:
Are you sure it's `IllegalState`, and not "strange things happened, just ignore it"?

Reply:
I think it's more about what we do with such situations right now.
Possibly yes, it's better just to log it.
I'm not sure

Reply:
Or applying suggested logic from the `setStateFunc` from VStateReport?

Reply:
I suggest to keep panic here if the author believes that it doesn't suppose to happen. Maybe `throw.Impossible()` would be even better if this is a case.